frontend/src/actions/Download.js

```diff
@@ -43,6 +45,18 @@ class Download extends Action {
         let directory = items.filter(items => items.directory === true);
         let fileItems = items.filter(items => items.file === true);
 
+        // String builder (singular/plural) for progress bar message
+        let msg = "";
+        let msgStart = "Downloading ";
+        let bothTypes = directory.length > 0 && fileItems.length > 0 ? true : false ;
+        let folderCount = directory.length > 0 ? (directory.length == 1 ? (directory.length+" folder") : (directory.length+" folders")) : ("");
```

Comment:
This is quiet difficult to read 
Maybe improve this by writing a Pluralize module in the `utils` directory of our project and use a plural function (example of how I wrote one)

```js
// frontend/src/utils/Pluralize.js
export function plural(word, amount, suffix = 's'){
  if(amount > 1){
    return word+suffix;
  }
  else{
    return word;
  }
}
```

and then use it like this
```suggestion
let folderCount = directory.length > 0 ? plural(`${directory.length} folder`, directory.length) : ("")
```

In general instead of doing addition chains of strings use [templated strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)
so for example 
```js
let msg = `Downloading ${folderCount}${msgMiddle}${filesCount}`
```
Maybe a tag function could be even cleaner to read:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#tagged_templates

Reply:
Thanks for the idea ! :) 
Already thought it would look ugly but couldn't think of a smoother way.

Yeah I'm not so sure if we can do anything about the progress of the time when it appears in the download folder.